src/dbt_osmosis/core/column_level_knowledge_propagator.py

```diff
@@ -84,3 +87,51 @@ def get_node_columns_with_inherited_knowledge(
         family_tree = _build_node_ancestor_tree(manifest, node)
         knowledge = _inherit_column_level_knowledge(manifest, family_tree, placeholders)
         return knowledge
+
+    @staticmethod
+    def update_undocumented_columns_with_prior_knowledge(
+        undocumented_columns: Iterable[str],
+        node: ManifestNode,
+        yaml_file_model_section: Dict[str, Any],
+        knowledge: Knowledge,
+        skip_add_tags: bool,
+        skip_merge_meta: bool,
+        add_progenitor_to_meta: bool,
+    ) -> int:
+        """Update undocumented columns with prior knowledge in node and model simultaneously
+        THIS MUTATES THE NODE AND MODEL OBJECTS so that state is always accurate"""
+        inheritables = ["description"]
+        if not skip_add_tags:
+            inheritables.append("tags")
+        if not skip_merge_meta:
+            inheritables.append("meta")
+
+        changes_committed = 0
+        for column in undocumented_columns:
+            prior_knowledge: ColumnLevelKnowledge = get_prior_knowledge(knowledge, column)
+            progenitor = prior_knowledge.pop("progenitor", None)
+            prior_knowledge = {k: v for k, v in prior_knowledge.items() if k in inheritables}
+            if add_progenitor_to_meta and progenitor:
```

Comment:
Moved forward from the original code. Because the original code mutates MODEL but did not mutate NODE.

- ref: https://github.com/z3z1ma/dbt-osmosis/blob/main/src/dbt_osmosis/core/osmosis.py#L1036-L1038